modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,595 @@
+
+					// If we don't have an attachment URL, don't include this image
+					if ( ! $attachment_url ) {
+						unset( $url['image:image'] );
+						continue;
+					}
+
+					$url['image:image']['loc'] = esc_url( $attachment_url );
+				}
+
+				// Only include title if not empty.
+				/** This filter is documented in wp-includes/feed.php */
+				if ( $attachment_title = apply_filters( 'the_title_rss', $attachment->post_title ) ) {
+					$url['image:image']['title'] = html_entity_decode( esc_html( $attachment_title ), ENT_XML1 );
+				}
+
+				// Only include caption if not empty.
+				/** This filter is documented in wp-includes/feed.php */
+				if ( $attachment_caption = apply_filters( 'the_excerpt_rss', $attachment->post_excerpt ) ) {
+					$url['image:image']['caption'] = html_entity_decode( esc_html( $attachment_caption ), ENT_XML1 );
+				}
+			}
+		}
+
+		if ( $post->post_modified_gmt && $post->post_modified_gmt != '0000-00-00 00:00:00' ) {
+			$post_latest_mod = $post->post_modified_gmt;
+		}
+		if ( $post->comment_count > 0 ) {
+			// last modified based on last comment
+			$latest_comment_datetime = jetpack_get_approved_comments_max_datetime( $post->ID );
+			if ( ! empty( $latest_comment_datetime ) ) {
+				if ( is_null( $post_latest_mod ) || $latest_comment_datetime > $post_latest_mod ) {
+					$post_latest_mod = $latest_comment_datetime;
+				}
+			}
+			unset( $latest_comment_datetime );
+		}
+		if ( ! empty( $post_latest_mod ) ) {
+			$latest_mod     = max( $latest_mod, $post_latest_mod );
+			$url['lastmod'] = jetpack_w3cdate_from_mysql( $post_latest_mod );
+		}
+		unset( $post_latest_mod );
+		if ( $post->post_type == 'page' ) {
+			$url['changefreq'] = 'weekly';
+			$url['priority']   = '0.6'; // set page priority above default priority of 0.5
+		} else {
+			$url['changefreq'] = 'monthly';
+		}
+		/**
+		 * Filter associative array with data to build <url> node and its descendants for current post.
+		 *
+		 * @since 3.9
+		 *
+		 * @param array $url Data to build parent and children nodes for current post.
+		 * @param int $post_id Current post ID.
+		 */
+		jetpack_sitemap_array_to_simplexml( array ( 'url' => apply_filters( 'jetpack_sitemap_url', $url, $post->ID ) ), $tree );
+		unset( $url );
+	}
+	$blog_home = array (
+		'loc'        => esc_url( get_option( 'home' ) ),
+		'changefreq' => 'daily',
+		'priority'   => '1.0'
+	);
+	if ( ! empty( $latest_mod ) ) {
+		$blog_home['lastmod'] = jetpack_w3cdate_from_mysql( $latest_mod );
+		header( 'Last-Modified:' . mysql2date( 'D, d M Y H:i:s', $latest_mod, 0 ) . ' GMT' );
+	}
+	/**
+	 * Filter associative array with data to build <url> node and its descendants for site home.
+	 *
+	 * @since 3.9
+	 *
+	 * @param array $blog_home Data to build parent and children nodes for site home.
+	 */
+	jetpack_sitemap_array_to_simplexml( array ( 'url' => apply_filters( 'jetpack_sitemap_url_home', $blog_home ) ), $tree );
```

Comment:
It might be best to break the filter onto its own line, to make things more readable and to make sure the parser can access all the information.